lib/nodelist.js

```diff
@@ -1,29 +1,28 @@
 define(['sorttable', 'snabbdom', 'helper'], function (SortTable, V, helper) {
   'use strict';
   V = V.default;
-
-  function getUptime(now, d) {
-    if (d.is_online && 'uptime' in d) {
-      return Math.round(d.uptime);
-    } else if (!d.is_online && 'lastseen' in d) {
-      return Math.round(-(now.unix() - d.lastseen.unix()));
-    }
-    return 0;
-  }
+  var now = new Date().getTime();
 
   function showUptime(uptime) {
-    var s = '';
-    uptime /= 3600;
+    var s = (now - uptime) / (60 * 1000);
 
-    if (uptime !== undefined) {
-      if (Math.abs(uptime) >= 24) {
-        s = Math.round(uptime / 24) + 'd';
-      } else {
-        s = Math.round(uptime) + 'h';
+    if (s !== undefined) {
+      if (Math.abs(s) < 60) {
+        return Math.round(s) + 'm';
+      }
+      s /= 60;
+      if (Math.abs(s) < 24) {
+        return Math.round(s) + 'h';
+      }
+      s /= 24;
+      if (Math.abs(s) < 365) {
+        return Math.round(s) + 'd';
       }
+      s /= 365;
+      return Math.round(s) + 'y';
```

Comment:
Do we need year in nodelist for uptime? nearly 3 years without firmware update or reboot